### ISSUE TITLE
TransformerView in multidimensional array

### ISSUE THREAD

**Issue by btaens** [andrej-griniuk/cakephp-fractal-transformer-view#2]
Hello,
Very useful plugin, works very nicely.

The only problem I'm having is with multidimensional arrays.
Let's say I have an entity named `Directory`.
I want to return 3 elements for serialization: `data`, `meta`, and `code`.

`data` is just an array of `Directory` entities, and works just fine.
`code` is a simple response code, no transformation necessary.
`meta`, however, is an array of miscellaneous data that is needed for the workings of the app, such as pagination, but also `crumbs`.

`crumbs` is an array of `Directory` entities, the breadcrumbs back to the main directory. This however, doesn't get transformed the way `data` does, as it is a sub-array of `meta`, and doesn't get checked by the transformer view.

Basically, there should be some solution to point to objects within arrays as transformable.

**Comment by andrej-griniuk** [comment]
Hi @btaens,

that totally makes sense. Will have a think about it... Thanks.

**Comment by andrej-griniuk** [comment]
Hi @btaens,
what you could do is create a transformer for `meta` and use fractal includes to transform `crumbs` or whatever you want in it. Something like:

``` php
class MetaTransformer extends TransformerAbstract
{
    protected $defaultIncludes = [
        'crumbs'
    ];

    /**
     * Creates a response item for each instance
     *
     * @param Product $product post entity
     * @return array transformed post
     */
    public function transform($meta)
    {
        return $meta;
    }

    public function includeCrumbs($meta)
    {
        $crumbs = $meta['crumbs'];

        return $this->collection($crumbs, new DirectoryTransformer());
    }
} 
```

Hope this helps.
